File: src/main/java/org/railwaystations/rsapi/app/auth/AuthUser.java
package org.railwaystations.rsapi.app.auth;

import jakarta.validation.constraints.NotNull;
import org.apache.commons.lang3.StringUtils;
import org.railwaystations.rsapi.core.model.User;
import org.springframework.security.core.GrantedAuthority;

import java.io.Serial;
import java.util.Collection;

public class AuthUser extends org.springframework.security.core.userdetails.User {

    @Serial
    private static final long serialVersionUID = 1L;

    private final User user;

    public AuthUser(@NotNull User user, Collection<? extends GrantedAuthority> authorities) {
        super(user.getDisplayName(), StringUtils.defaultString(user.getKey()), authorities);
        this.user = user;
    }

    public User getUser() {
        return user;
    }
}

File: adapter/src/main/kotlin/org/railwaystations/rsapi/adapter/web/auth/AuthUser.kt
package org.railwaystations.rsapi.adapter.web.auth

import jakarta.validation.constraints.NotNull
import org.railwaystations.rsapi.core.model.User
import org.springframework.security.core.GrantedAuthority

class AuthUser(val user: @NotNull User, authorities: Collection<GrantedAuthority>) :
    org.springframework.security.core.userdetails.User(
        user.name, user.key ?: "", authorities
    )
